python/GafferSceneUI/InteractiveRenderUI.py

```diff
@@ -36,6 +36,70 @@
 
 import Gaffer
 import GafferScene
+import GafferUI
+
+##########################################################################
+# UI for the state plug that allows setting the state through buttons
+##########################################################################
+
+from collections import namedtuple
+
+ButtonLabels = namedtuple( 'ButtonLabels', 'startButton pauseButton')
+
+class InteractiveRenderingStateWidget( GafferUI.PlugValueWidget ) :
+
+	labelMap = {
+		GafferScene.InteractiveRender.State.Stopped : ButtonLabels('Start', 'Pause'),
+		GafferScene.InteractiveRender.State.Running : ButtonLabels('Stop', 'Pause'),
+		GafferScene.InteractiveRender.State.Paused : ButtonLabels('Continue', 'Paused') }
+
+	def __init__( self, plug, *args, **kwargs) :
+
+		self.plug = plug
+
+		row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+		GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+		with row :
+			self.startButton = GafferUI.Button()
+			self.pauseButton = GafferUI.Button()
+
+			self.startConnection = self.startButton.clickedSignal().connect( Gaffer.WeakMethod( self._startStopCallback ) )
+			self.pauseConnection = self.pauseButton.clickedSignal().connect( Gaffer.WeakMethod( self._pauseCallback ) )
+
+		self._updateButtons()
+
+	def _updateFromPlug( self ) :
+		pass
+
+	def hasLabel( self ) :
+		return True
+
+	def _isRunning( self ) :
```

Comment:
Actually, I think it's more than a taste thing. The usage of these methods includes things like `self._isStopped() or self._isPaused()`, so the utility methods are designed such that two calls to `getValue()` are required where one would be sufficient. I think the following pattern is preferable (also taking into account my comment about contexts) :

```
with self.getContext() :
    state = self.getPlug().getValue()

if state == GafferScene.InteractiveRender.State.Stopped or state == GafferScene.InteractiveRender.State.Paused :
    ...

```

Now, I admit that those verbose accesses to the State enum are a drag, so I can understand the motivation to hide them somehow. But I think the code structure is more important in this case, and if you wanted to you could use an alias for `GafferScene.InteractiveRender.State` you could improve readability without changing the structure.